engine.py

```diff
@@ -499,9 +498,24 @@ def _create_dialog(self, title, bundle, widget, parent):
         :param parent: The parent QWidget for the dialog
         """
 
+        from tank.platform.qt import QtCore
+
         # call the base implementation to create the dialog:
         dialog = tank.platform.Engine._create_dialog(self, title, bundle, widget, parent)
 
+        if dialog.parent():
+            # parenting crushes the dialog's style. This seems to work to reset
+            # the style to the dark look and feel in preparation for the
+            # re-application below.
```

Comment:
i feel it is important to explain why we are doing all this. Maybe we could add a comment to point to the big explanation next to `_initialize_dark_look_and_feel`? Just to help future maintainers of the code understand the intent.